src/actions/diagnostics.rs

```diff
@@ -31,10 +34,8 @@ pub struct Suggestion {
 }
 
 #[derive(Debug)]
-pub struct FileDiagnostic {
-    pub file_path: PathBuf,
-    pub main: (Diagnostic, Vec<Suggestion>),
-    pub secondaries: Vec<(Diagnostic, Vec<Suggestion>)>,
+pub struct ParsedDiagnostics {
+    pub diagnostics: HashMap<PathBuf, Vec<(Diagnostic, Vec<Suggestion>)>>,
```

Comment:
The reason for this split wrt paths is that theoretically a single compiler message it can have multiple primary spans that point to separate files